### ISSUE TITLE
Suggestion: autoclose open file streams upon destruction (optional argument?)

### ISSUE THREAD

**Issue by iwelch** [JuliaLang/julia#26476]
**Suggestion**: (Add an optional argument to automatically guarantee to) close files (IOStreams) upon destruction.

Come to think of it, should this not be default behavior?  are there ever any use cases where a file should stay open after access to it has become impossible?  (if so, maybe there could be an option not to autoclose?)

this change would validate such constructs as `serialize( open("m.bin", "w"), x )`.  incidentally, this already works, but with the closing, it would be guaranteed to work.

I do appreciate the existing methods to do this, too, like open(function,...) and open() do ... enddo.

**Comment by JeffBezanson** [comment]
I believe we already add `close` as a finalizer for file streams. When this actually happens depends on when GC runs though.

**Comment by StefanKarpinski** [comment]
Yes, this already is the behavior. Related feature issue: https://github.com/JuliaLang/julia/issues/7721.

**Comment by iwelch** [comment]
excellent.  (add to docs?)

**Comment by StefanKarpinski** [comment]
Sure, if you can figure out a good place to mention this, please do make a PR adding it.

**Comment by iwelch** [comment]
sorry: what's a PR?

I would stick it into https://docs.julialang.org/en/stable/stdlib/io-network/#Base.open : "Open a file in a mode specified by five boolean arguments. The default is to open files for reading only. Returns a stream for accessing the file.  *Files are automatically closed when the IOStream is garbage collected*."

**Comment by StefanKarpinski** [comment]
PR = pull request: https://help.github.com/articles/about-pull-requests/
